Packet++/header/Sll2Layer.h

```diff
@@ -0,0 +1,116 @@
+#ifndef PACKETPP_SLL2_LAYER
+#define PACKETPP_SLL2_LAYER
+
+#include "MacAddress.h"
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+	/**
+	 * @struct sll2_header
+	 * Represents SLL2 header
+	 */
+#pragma pack(push, 1)
+	struct sll2_header
+	{
+		/** Contains an Ethernet protocol type of the next layer */
+		uint16_t protocol_type;
+		/** The "Reserved (MBZ)" field is reserved, and must be set to zero */
+		uint16_t reserved_type;
+		/** The interface index field is a signed integer in network byte 
+		 * order and contains the 1-based index of the interface on which the packet was observed 
+		 **/
+		int32_t interface_index_type;
```

Comment:
shouldn't this be `uint32_t`?

Also: shouldn't the name just be `interface_index`?

Reply:
https://www.tcpdump.org/linktypes/LINKTYPE_LINUX_SLL2.html
doc seems  it's a signed integer.
name can be changed properly.

Reply:
maybe official doc gave the wrong description. 🤔

Reply:
let's just make it `uint32_t`?